cmd/soroban-cli/src/commands/contract/arg_parsing.rs

```diff
@@ -215,24 +219,40 @@ fn parse_single_argument(
         ) {
             let trimmed_s = s.trim_matches('"');
             let addr = resolve_address(trimmed_s, config)?;
-            if let Some(signer) = resolve_signer(trimmed_s, config) {
+            if let Some(signer) = resolve_signer(trimmed_s, config).await {
```

Comment:
we should do this outside of argument parsing so it only occurs once.

In testing, I was getting prompted multiple times for access to my key before the CLI attempted to simulate, and I assume this is the culprit

Reply:
This is a good catch. I think I'd like to handle this in a separate PR though, because I think that being prompted multiple times by keychain for access to the key could be related to something else instead/in addition to this.

I was previously seeing it on key generation, and fetching the public key... and now that isn't consistently happening. So i think this is worthy of a separate investigation.